.github/workflows/reusable-linters.yml

```diff
@@ -71,9 +71,7 @@ jobs:
       with:
         token: ${{ secrets.codecov-token }}
         files: >-
-          .tox/.tmp/.mypy/python-3.12/cobertura.xml,
-          .tox/.tmp/.mypy/python-3.10/cobertura.xml,
-          .tox/.tmp/.mypy/python-3.8/cobertura.xml
+          .tox/.tmp/.mypy/python-3.11/cobertura.xml
```

Comment:
We should be running against a matrix of Pythons..

Reply:
I've highlighted this every time someone tries to add a matrix for mypy. The extra Python versions never seem to catch any extra errors the user would encounter, but as we introduce more strict typing the internal errors in the codebase differ slightly, making it pretty much impossible to get mypy passing on all versions simultaneously.